pkg/util/kube/pki.go

```diff
@@ -33,26 +32,17 @@ import (
 // secret with 'name' in 'namespace'. It will read the private key data from the secret
 // entry with name 'keyName'.
 func SecretTLSKeyRef(ctx context.Context, secretLister corelisters.SecretLister, namespace, name, keyName string) (crypto.Signer, error) {
-	log := logf.FromContext(ctx)
-	log = logf.WithRelatedResourceName(log, name, namespace, "Secret")
-
 	secret, err := secretLister.Secrets(namespace).Get(name)
 	if err != nil {
-		log.Error(err, "failed to retrieve secret")
```

Comment:
Are we loosing information here?

Reply:
Nope, this is passed back as a IsNotFound error if it is not found, and handled appropriately. Otherwise it is/should be logged by the call-site.